database/rrdcalctemplate.c

```diff
@@ -45,6 +45,11 @@ static int rrdcalctemplate_is_there_label_restriction(RRDCALCTEMPLATE *rt,  RRDH
 }
 
 static inline int rrdcalctemplate_test_additional_restriction(RRDCALCTEMPLATE *rt, RRDSET *st) {
+    if (rt->charts_pattern &&
+        !simple_pattern_matches(rt->charts_pattern, st->id) &&
+         !simple_pattern_matches(rt->charts_pattern, st->name))
```

Comment:
I think it should be _or_ (_if pattern and (!matches_id || 1matches_name)_)

Reply:
All right, I tested your suggestion with this alarm:

``` 
      on: disk.svctm
    calc: $last_collected_t
   units: boolean
  charts: !*sdc* *
   every: 1m
    warn: $this != nan || $this == nan
    info: testing with disks
      to: sysadmin
```

And everything works.